gocov/convert/convert.go

```diff
@@ -74,10 +58,35 @@ func ConvertProfiles(filenames ...string) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
-		for _, p := range profiles {
-			if err := converter.convertProfile(packages, localModuleName, p); err != nil {
+
+		sort.Slice(profiles, func(i, j int) bool {
+			return profiles[i].FileName < profiles[j].FileName
+		})
+
+		//packageNameGoPkgsMap := make(map[string][]*goPackages.Package)
+		//profilePkgsMap := make(map[*cover.Profile][]*goPackages.Package)
+		tempCache := make(map[string][]*goPackages.Package)
+		var prevPackage string
```

Comment:
I think you could change this to:

```go
var module *goPackages.Module
```

Then in the loop, you can check if the previously loaded module path is a prefix of the package path. i.e.

```go
		for _, profile := range profiles {
			packageName := filepath.Dir(profile.FileName)
			if module == nil || !strings.HasPrefix(packageName, module.Path) {
				pkgs, err := goPackages.Load(&goPackages.Config{
					Mode: goPackages.NeedName | goPackages.NeedModule,
				}, packageName)
				if err != nil {
					return nil, fmt.Errorf("loading packages for %s: %w", packageName, err)
				}
				module = pkgs[0].Module
			}
			if err := converter.convertPackage(module, profile); err != nil {
				return nil, err
			}
		}
```